driver/src/orderbook/streamed/balance.rs

```diff
@@ -2,17 +2,43 @@ use super::*;
 use anyhow::{anyhow, ensure, Result};
 use serde::{Deserialize, Serialize};
 
+/// Balance change from a potential solution that might still get replaced by a better solution in
+/// the same batch.
+#[derive(Clone, Copy, Debug, Default, Deserialize, Serialize)]
+struct Proceeds {
+    batch_id: BatchId,
+    // Two fields instead of one I256 because although unlikely it could be possible to overflow a
+    // I256 with multiple trades in one batch.
```

Comment:
Maybe we can consider using a `BigInt` library so we don't have to deal with math overflows in places where they are allowed.

Reply:
I also find a single variable simpler, if possible. This way we wouldn't need the `getField` method and only have to do one addition in `balance_with_deposit_and_proceeds`.

Reply:
That is a neat idea. If we used a BigInt everywhere (not just proceeds) for balances we could remove all of the overflow checks. Then we would only need to check whether the value really fits in a U256 when actually creating the account states.

Since we have this code now and with this PR the balance code should be done and its already reviewed we should keep this as is now and I will keep this as a todo for when the event based orderbook has been meged into master.